lab_1/main.py

```diff
@@ -87,27 +171,52 @@ def get_adjacent_words(tokens: list, word: str, left_n: int, right_n: int) -> li
     right_n = 3
     --> [['man', 'is'], ['dog, 'cat']]
     """
-    pass
-
-
-def read_from_file(path_to_file: str) -> str:
+    adjacent_words = []
+    words = []
+    if type(tokens) == list and type(word) == str and type(left_n) == int and type(right_n) == int:
+        if left_n <= 0 and right_n > 0\
+                or left_n > 0 and right_n <= 0\
+                or left_n > 0 and right_n > 0:
+
+            operative_tokens = get_concordance(tokens, word, left_n, right_n)
+            for i in operative_tokens:
+                word = word.lower()
+                i.remove(word)
+                if len(i) > 2:
+                    words.append(i[0])
+                    words.append(i[-1])
+                    adjacent_words.append(words[:])
+                    words.clear()
+                else:
+                    adjacent_words.append(i)
+
+    return adjacent_words
+
+
+#7
+def write_to_file(path_to_file: str, content: list):
     """
-    Opens the file and reads its content
-    :return: the initial text in string format
+    Writes the result in a file
     """
-    with open(path_to_file, 'r', encoding='utf-8') as fs:
-        data = fs.read()
+    if type(path_to_file) == str and type(content) == list:
```

Comment:
Usually, there are check for input arguments and body function - on the same level.
No need for nesting.
Also, `type`.